gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/GemFireCacheImpl.java

```diff
@@ -1065,13 +1065,14 @@ public void run() {
             callbacks.getStoragePoolSize(true);
       }
       this.memorySize = memorySize;
-      if (memorySize > 0) {
+      if (memorySize > 0 && CallbackFactoryProvider.getStoreCallbacks().isEnterpriseEdition()) {
```

Comment:
> Why through callback?

GemFireVersion.isEnterpriseEdition loads the GemfireVersion.properties while the property we seek here is available in GemFireXDVersion.properties file. And GemFireXDVersion is not accessible from gemfire module.